src/System.Runtime.Extensions/src/System/IO/Path.WinRT.cs

```diff
@@ -11,12 +12,18 @@ namespace System.IO
 {
     public static partial class Path
     {
-        private static byte[] CreateCryptoRandomByteArray(int byteLength)
+        private static unsafe void GetCryptoRandomBytes(byte* bytes, int byteCount)
         {
+            Debug.Assert(bytes != null);
+            Debug.Assert(byteCount >= 0);
+
             byte[] arr;
-            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteLength);
+            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteCount);
             CryptographicBuffer.CopyToByteArray(buffer, out arr);
-            return arr;
+
+            Debug.Assert(arr.Length == byteCount);
+
+            Marshal.Copy(arr, 0, new IntPtr(bytes), byteCount);
```

Comment:
@justinvp Ah, ok. I hadn't realized that the result of `GenerateRandom` would always be castable to `IBufferByteAccess` (and that `GetBuffer` would return IntPtr instead of byte[]). In that case, I'd say it's ok to make the change.